### PR TITLE
Feature/dockerized pipeline

### PR DESCRIPTION
**Pull request by izcram**
Code to run pipeline in a docker container.

Needs some more fine tuning (WIP), but overall structure is there.

After the review, I'll squash the commits together, s.t. we don't have WIP commits in our master branch ;)

### CODE REVIEW

File: pipeline/docker/README.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+```
+docker run -u $(id -u ${USER}):$(id -g ${USER}) \
+       -v  path_to_resource_files:/files/resources \
+       -v  path_to_output_directory:/files/data \
+       -v  optional_path_to_code_base:/opt/brca-exchange \
+       -v  path_to_pipeline_credentials.cfg:/opt/luigi_pipeline_credentials.cfg \
+       -v  path_to_previous_release_built_with_change_types.tsv:/files/previous_release/built_with_change_types.tsv \
+       -v  path_to_release_notes.txt:/files/release_notes.txt \
+       -v /tmp:/.synapseCache \
+       -e PREVIOUS_RELEASE_DATE=07-20-2017 \
```
Comment (zfisch) [inline pipeline/docker/README.md]: `PREVIOUS_RELEASE_DATE=07-20-2017` should probably be something like `PREVIOUS_RELEASE_DATE=MM-DD-YYYY`

File: pipeline/docker/run_luigi.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+RELEASE_NOTES=/files/release_notes.txt
+
+cd /opt/brca-exchange
+
+echo "Running brca exchange pipeline:"
+echo "Git hash: $(git log | head -n 1)"
+
+cd /opt/brca-exchange/pipeline/luigi
+
+python -m luigi --module CompileVCFFiles RunAll --resources-dir ${BRCA_RESOURCES} --file-parent-dir ${PARENT_DIR} --otuput-dir ${OUTPUT_DIR} --previous-release ${PREVIOUS_RELEASE} --previous-release-date ${PREVIOUS_RELEASE_DATE} --release-notes ${RELEASE_NOTES} --local-scheduler
```
Comment (zfisch) [inline pipeline/docker/run_luigi.sh]: definitely want to include the fix for `--otuput-dir ${OUTPUT_DIR}` to `--output-dir ${OUTPUT_DIR}`

File: pipeline/requirements.txt [hunk 3]
```diff
@@ -5,26 +5,17 @@ bioutils==0.1.4
...
 lockfile==0.12.2
 luigi==2.2.0
 nose==1.3.7
 numpy==1.11.1
 Parsley==1.3
-pip==8.1.2
-psycopg2==2.6.2
+pip==9.0.1
 pygr==0.8.2
-pyhgvs==0.9.4
```
Comment (zfisch) [inline pipeline/requirements.txt]: this seems odd -- i know pyhgvs is used during brca_pseudonym_generator and some other places throughout the pipeline

Comment (melissacline) [reply pipeline/requirements.txt]: We might be able to get rid of this dependency in the future, now that Reese Hart's hgvs library converts to hg38 coordinates.  The reason why we were using pyhgvs (the Counsyl library) before was that the hgvs library didn't support hg38 yet.

Comment (izcram) [reply pipeline/requirements.txt]: 0.9.4 doesn't seem to be in PyPi, therefore it is built from source instead: https://github.com/BRCAChallenge/brca-exchange/pull/617/files#diff-8f09b8970829c955182f646b3026bccbR42

Good to know that we can get rid of it, I found it quite confusing to have hgvs and pyhgvs in the same code ;)

**Pull request merged**